tests/qepsi4/_psi4_test.py

```diff
@@ -48,199 +81,237 @@ def create_molecule(geometry, charge, mult):
+        [
+            (
+                Psi4Config(hydrogen_geometry, 1, options={"scf_type": "pk"}),
+                2,
+                ExpectedTuple(-0.8543376267387818, 1, 1, 2, 0, 1, 2),
+            ),
+            (
+                Psi4Config(hydrogen_geometry, 1, 0, options={"scf_type": "pk"}),
+                0,
+                ExpectedTuple(-0.8543376267387818, 1, 1, 2, 1, 0, 2),
+            ),
+            (
+                Psi4Config(hydrogen_geometry),
+                2,
+                ExpectedTuple(
+                    -0.8544322638069642, 1, 1, 2, 0, 0, 4, -0.9714266904819895
+                ),
+            ),
+            (
+                Psi4Config(dilithium_geometry, 4, 2,),
+                4,
+                ExpectedTuple(exp_n_qubits=8, extra_energy_check=-14.654620243980217),
+            ),
+            (
+                Psi4Config(hydrogen_geometry, method="fci", save_rdms=True),
+                2,
+                ExpectedTuple(None, 1, 1, 2, 0, 0, 4),
+            ),
+            (
+                Psi4Config(dilithium_geometry, 4, 1, method="fci", save_rdms=True),
+                2,
+                ExpectedTuple(exp_n_qubits=8, exp_one_body_tensor_shape=8),
+            ),
+            (
+                Psi4Config(
+                    dilithium_geometry,
+                    freeze_core=True,
+                    freeze_core_extract=True,
+                    method="fci",
+                    save_rdms=True,
+                ),
+                None,
+                ExpectedTuple(exp_n_qubits=16, exp_one_body_tensor_shape=16),
+            ),
+            (
+                Psi4Config(
+                    dilithium_geometry, freeze_core=True, freeze_core_extract=True
+                ),
+                None,
+                ExpectedTuple(exp_n_qubits=16),
+            ),
+        ],
+    )
+    def test_run_psi4(
+        self,
+        psi4_config: Psi4Config,
+        jw_wagner_particle_num: int,
+        expected_tuple: ExpectedTuple,
+    ):
+        results_dict = run_psi4(
+            geometry=psi4_config.geometry,
+            n_active_extract=psi4_config.n_active_extract,
+            n_occupied_extract=psi4_config.n_occupied_extract,
+            freeze_core=psi4_config.freeze_core,
+            freeze_core_extract=psi4_config.freeze_core_extract,
+            method=psi4_config.method,
+            options=psi4_config.options,
+            save_hamiltonian=True,
+            save_rdms=psi4_config.save_rdms,
+        )
 
-def test_run_psi4_2():
+        results, hamiltonian, rdm = (
+            results_dict["results"],
+            results_dict["hamiltonian"],
+            results_dict["rdms"],
+        )
 
-    options = {'scf_type' : 'pk'} # triggers exact ERI calculation
+        if rdm:
+            energy_from_rdm = rdm.expectation(hamiltonian)
+
+        if psi4_config.save_rdms:
+            math.isclose(results["energy"], energy_from_rdm)
```

Comment:
Is there a missing `assert` here?

Reply:
@max-radin Not really. The original assert that followed this statement is done later in the code (line #228, #282)

Reply:
Ok looking more closely at this, the logic for checking energy values is pretty convoluted. For example, the check on line 282 is skipped if `extra_energy_check` is provided, whereas the check on line 231 is skipped if `save_rdms` is true.

What do you think about reorganizing the energy checks as follows? I think this would be simpler and also more thorough than the original tests.
* Remove `extra_energy_check` and instead just use `exp_energy`.
* If `save_rdms` is true, then assert that  `energy_from_rdm` matches `results["energy"]`
* If particle number is provided, then assert that `energy` matches `results["energy"]`